game-core/src/main/java/games/strategy/triplea/ui/PoliticsPanel.java

```diff
@@ -57,61 +55,81 @@
         SwingAction.of(
             "Do Politics...",
             e -> {
-              final Dimension screenResolution = Toolkit.getDefaultToolkit().getScreenSize();
-              final int availHeight = screenResolution.height - 96;
-              final int availWidth = screenResolution.width - 30;
-              final int availHeightOverview = (int) ((float) availHeight * 2 / 3);
-              final int availHeightChoice = (int) ((float) availHeight / 3);
-
               final JDialog politicalChoiceDialog = new JDialog(parent, "Political Actions", true);
               final Insets insets = new Insets(1, 1, 1, 1);
               final JPanel politicalChoicePanel = new JPanel();
               politicalChoicePanel.setLayout(new GridBagLayout());
               final PoliticalStateOverview overview =
-                  new PoliticalStateOverview(getData(), getMap().getUiContext(), false);
-              final JScrollPane overviewScroll = new JScrollPane(overview);
-              overviewScroll.setBorder(BorderFactory.createEmptyBorder());
-              // add 26 to height when the actions are empty, because for some stupid reason java
-              // calculates the pack size wrong
-              // (again)...
-              // add 20 to either when the opposite needs scroll bars, because that is how big
-              // scroll bars are..
-              overviewScroll.setPreferredSize(
-                  new Dimension(
-                      (overviewScroll.getPreferredSize().width > availWidth
-                          ? availWidth
-                          : (overviewScroll.getPreferredSize().width
-                              + (overviewScroll.getPreferredSize().height > availHeightOverview
-                                  ? 20
-                                  : 0))),
-                      (overviewScroll.getPreferredSize().height > availHeightOverview
-                          ? availHeightOverview
-                          : (overviewScroll.getPreferredSize().height
-                              + (validPoliticalActions.isEmpty() ? 26 : 0)
-                              + (overviewScroll.getPreferredSize().width > availWidth ? 20 : 0)))));
+                  new PoliticalStateOverview(getData(), getMap().getUiContext(), false) {
+
+                    private GridBagConstraints makeGridBagConstraints(
```

Comment:
For consideration, there is a 'GridBagConstraintBuilder' that might be of help to you here.